src/ast/mod.rs

```diff
@@ -321,7 +345,37 @@ where
         name: Symbol,
         value: GenericExpr<Head, Leaf>,
     },
-    /// Declare a user-defined datatype.
+
+    /// Create a new user-defined sort, which can then
+    /// be used in new [`Command::Function`] declarations.
+    /// The [`Command::Datatype`] command desugars directly to this command, with one [`Command::Function`]
+    /// per constructor.
+    /// The main use of this command (as opposed to using [`Command::Datatype`]) is for forward-declaring a sort for mutually-recursive datatypes.
+    ///
+    /// It can also be used to create
+    /// a container sort.
+    /// For example, here's how to make a sort for vectors
+    /// of some user-defined sort `Math`:
+    /// ```text
+    /// (sort MathVec (Vec Math))
+    /// ```
+    ///
+    /// Now `MathVec` can be used as an input or output sort.
+    Sort(Span, Symbol, Option<(Symbol, Vec<Expr>)>),
+
+    /// Egglog supports three types of functions
+    ///
+    /// A constructor models an egg-style user-defined datatype
+    /// It can only be defined through the `datatype`/`datatype*` command
+    /// or the `constructor` command
+    ///
+    /// A relation models a datalog-style mathematical relation
+    /// It can only be defined through the `relation` command
+    ///
+    /// A custom function is a map
```

Comment:
The map part of the definition is a bit weird to me, but it's fine for now since we will need a big documentation refactor anyway.